File: client/java/src/main/java/com/be/mesh/client/system/MeshRegistry.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.system;

import com.be.mesh.client.annotate.SPI;
import com.be.mesh.client.mpc.ServiceProxy;
import com.be.mesh.client.prsim.Registry;
import com.be.mesh.client.struct.Registration;

import java.util.List;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public class MeshRegistry implements Registry {

    private final Registry registry = ServiceProxy.proxy(Registry.class);


    @Override
    public void register(Registration registration) {
        registry.register(registration);
    }

    @Override
    public void registers(List<Registration> registrations) {
        registry.registers(registrations);
    }

    @Override
    public void unregister(Registration registration) {
        registry.unregister(registration);
    }

    @Override
    public List<Registration> export(String kind) {
        return registry.export(kind);
    }
}

File: client/golang/system/mesh_registry.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package system

import (
	"context"
	"github.com/opendatav/mesh/client/golang/macro"
	"github.com/opendatav/mesh/client/golang/prsim"
	"github.com/opendatav/mesh/client/golang/types"
)

func init() {
	var _ prsim.Registry = new(systemRegistry)
	macro.Provide(prsim.IRegistry, new(systemRegistry))
}

type systemRegistry struct {
}

func (that *systemRegistry) Att() *macro.Att {
	return &macro.Att{Name: macro.MeshSys}
}

func (that *systemRegistry) Register(ctx context.Context, registration *types.Registration[any]) error {
	return aware.Registry.Register(ctx, registration)
}

func (that *systemRegistry) Registers(ctx context.Context, registrations []*types.Registration[any]) error {
	return aware.Registry.Registers(ctx, registrations)
}

func (that *systemRegistry) Unregister(ctx context.Context, registration *types.Registration[any]) error {
	return aware.Registry.Unregister(ctx, registration)
}

func (that *systemRegistry) Export(ctx context.Context, kind string) ([]*types.Registration[any], error) {
	return aware.Registry.Export(ctx, kind)
}
